packages/react-components/src/components/TextFieldWithMention/TextFieldWithMention.tsx

```diff
@@ -0,0 +1,629 @@
+      const newPlainText =
+        inputTextValue.substring(0, currentTriggerStartIndex) + mention + inputTextValue.substring(selectionEnd);
+      const triggerText = mentionLookupOptions?.trigger ?? DEFAULT_MENTION_TRIGGER;
+      // update html text with updated plain text
+      const updatedContent = updateHTML({
+        htmlText: textValue,
+        oldPlainText,
+        newPlainText,
+        tags: tagsValue,
+        startIndex: currentTriggerStartIndex,
+        oldPlainTextEndIndex: selectionEnd,
+        change: mention,
+        mentionTrigger: triggerText
+      });
+      const displayName = getDisplayNameForMentionSuggestion(suggestion, localeStrings);
+      const newCaretIndex = currentTriggerStartIndex + displayName.length + triggerText.length;
+      // move the caret in the text field to the end of the mention plain text
+      setCaretIndex(newCaretIndex);
+      setSelectionEndValue(newCaretIndex);
+      setSelectionStartValue(newCaretIndex);
+      setCurrentTriggerStartIndex(-1);
+      updateMentionSuggestions([]);
+      // set focus back to text field
+      textFieldRef?.current?.focus();
+      setActiveSuggestionIndex(undefined);
+      onChange && onChange(undefined, updatedContent.updatedHTML);
+    },
+    [
+      textFieldRef,
+      inputTextValue,
+      currentTriggerStartIndex,
+      mentionLookupOptions?.trigger,
+      onChange,
+      textValue,
+      tagsValue,
+      updateMentionSuggestions,
+      localeStrings
+    ]
+  );
+
+  const onTextFieldKeyDown = useCallback(
+    (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      // caretIndex should be set to undefined when the user is typing
+      setCaretIndex(undefined);
+      // shouldHandleOnMouseDownDuringSelect should be set to false after the last mouse down event.
+      // it shouldn't be updated in onMouseUp
+      // as onMouseUp can be triggered before or after onSelect event
+      // because its order depends on mouse events not selection.
+      setShouldHandleOnMouseDownDuringSelect(false);
+      if (isEnterKeyEventFromCompositionSession(ev)) {
+        return;
+      }
+      if (ev.key === 'ArrowUp') {
+        ev.preventDefault();
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? mentionSuggestions.length - 1
+              : Math.max(activeSuggestionIndex - 1, 0);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      } else if (ev.key === 'ArrowDown') {
+        ev.preventDefault();
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? 0
+              : Math.min(activeSuggestionIndex + 1, mentionSuggestions.length - 1);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      }
+      if (ev.key === 'Enter' && (ev.shiftKey === false || !supportNewline)) {
+        ev.preventDefault();
```

Comment:
Should we have the preventDefault() function in the condition below, or no?

Reply:
> Should we have the preventDefault() function in the condition below, or no?

Will wrap it in the condition below